### PR TITLE
rm hardcoded mocked tx

### PR DESCRIPTION
**Pull request by tcoratger**
Related to https://github.com/kkrt-labs/kakarot-rpc/pull/1324, we can now recover a signer properly from the arbitrary transactions generated for tests. As a result, we don't need hardcoded values anymore.

This PR is responsible for removing constants and hardcoded values ​​that are no longer useful.

### CODE REVIEW

File: tests/tests/debug_api.rs [hunk 1]
```diff
@@ -44,117 +44,34 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
-        *BLOCK_HASH,
-        BLOCK_NUMBER,
-        None,
-        0,
-    );
+    // Checking some fields but not all since mocked signer is not exact at the moment
+    assert_eq!(transaction.block_hash, header.hash);
+    assert_eq!(transaction.block_number, header.number);
+    assert_eq!(transaction.nonce, tx.nonce);
+    assert_eq!(transaction.chain_id, tx.chain_id);
```
Comment (tcoratger) [inline tests/tests/debug_api.rs:60]: @greged93 Here is the only place in the PR where we slightly lose information but I think that with what I did here it is equivalent:
- As we now use randomly generated transactions, we no longer need to do a test per transaction type, we are quite covered with a single call (moreover, the conversion with the types is handled by reth so we do not need to do as many tests).
- Where we are slightly losing is that the signing of the tx is not correct as you pointed out in your comments here https://github.com/kkrt-labs/kakarot-rpc/pull/1324#issuecomment-2278814099, so we can no longer use the approach aimed at reconstructing the transaction with the Bytes to compare it to the one obtained by get transaction by hash because some elements such as the hash for example which is recalculated do not match. To do this I adopted the following approach:
    - I made the first call with `debug_getRawTransaction`, then I rebuilt the transaction despite the fact that the signer was not the right one.
    - I then compare a few fields that I know are correct with my Katana transaction that I have in memory.
    - I redo my second test with `eth_getTransactionByHash` and this time I compare my full result with my Katana transaction that I have in memory.
    - So I think we are sufficiently covered here, especially since the conversion parts are handled by reth so we should not need to test everything.

I did all this to try to simplify our test environment as much as possible and remove all the constants and heavy functions that were only there for this test. Tell me what you think but I think the coverage remains largely reasonable here. And if we are not 100% happy, in follow up, we can always improve our mocking to try to have an exact transaction with the right signer.

Comment (tcoratger) [reply tests/tests/debug_api.rs]: @greged93 Now this should be good, I used the reth test utils `sign_tx_with_random_key_pair` method to sign the random transaction in the arbitrary generation. I will fix the `arbitrary_with_optional_fields` to `arbitrary` in a follow up PR as there are several places to change in the code and it is not totally related to this PR which is already pretty big to review :)

File: src/providers/eth_provider/database/types/transaction.rs [hunk 2]
```diff
@@ -254,4 +223,36 @@ mod tests {
 
         let _ = StoredTransaction::arbitrary(&mut arbitrary::Unstructured::new(&bytes)).unwrap();
     }
+
+    #[test]
+    fn random_tx_signature() {
+        for _ in 0..100 {
```
Comment (greged93) [inline src/providers/eth_provider/database/types/transaction.rs]: I think we can just gave one try

File: src/test_utils/mongo/mod.rs [hunk 3]
```diff
@@ -156,9 +128,11 @@ impl MongoFuzzer {
     }
 
     /// Adds a transaction to the collection of transactions with custom values.
-    pub fn add_custom_transaction(&mut self, builder: TransactionBuilder) -> Result<(), Box<dyn std::error::Error>> {
-        // Build a transaction using the provided builder and random byte size.
-        let transaction = builder.build(self.rnd_bytes_size)?;
+    pub fn add_custom_transaction(&mut self) -> Result<(), Box<dyn std::error::Error>> {
```
Comment (greged93) [inline src/test_utils/mongo/mod.rs:131]: we should probably remove this and just keep `add_random_transactions`, what do you think?

File: tests/tests/debug_api.rs [hunk 4]
```diff
@@ -44,116 +44,40 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
+        transaction.signature = Some(reth_rpc_types::Signature {
+            y_parity: rpc_transaction.signature.unwrap().y_parity,
+            ..transaction.signature.unwrap()
+        });
+    }
+
+    // For EIP1559, `gas_price` is recomputed as per https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L54
+    if rpc_transaction.transaction_type.unwrap() == 2 {
+        transaction.gas_price = rpc_transaction.gas_price;
+    }
```
Comment (greged93) [inline tests/tests/debug_api.rs:79]: Why do we need this? Does this perhaps indicate that we are doing something wrong elsewhere?

File: src/test_utils/mongo/mod.rs (hunk 3 above)
Comment (tcoratger) [reply src/test_utils/mongo/mod.rs]: agreed, fixed

File: tests/tests/debug_api.rs (hunk 4 above)
Comment (tcoratger) [reply tests/tests/debug_api.rs:79]: Honestly I don't understand these calculations in reth and I don't feel like they apply to us. In fact it is in the `from_recovered_with_block_context` function that these strange calculations are performed:
- For the parity, in the function here https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/signature.rs#L17, the parity is set to `None` for Legacy transactions. But on the one hand, in the arbitrary definition of transactions, if I set to `None` for Legacy transactions then the signature test is no longer valid. On the other hand I don't understand why the parity would be `None` because it does exist and is also used to calculate the `v` using the chain ID too...
- For the `gas_price`, I have the impression that they also perform a recalculation of the gas price here https://github.com/paradigmxyz/reth/blob/603e39ab74509e0863fc023461a4c760fb2126d1/crates/rpc/rpc-types-compat/src/transaction/mod.rs#L54 with the tip and the base fee of the block but I don't have the impression that it is applicable for us right?

So in conclusion I just filled these two fields with the solutions to pass the test because I don't have the impression that it is something that matters to us. What do you think?

Comment (greged93) [reply tests/tests/debug_api.rs:79]: I was actually looking at the code for the parity and also feel like the parity should only be set for transactions that aren't legacy, since for legacy transactions, you can recover the parity by doing `(v - 35)%2` which means you don't need the parity right? What I mean by this is that for legacy transactions, the parity is encoded in the `v` field.

For the gas price I am not too sure, we can leave it like this I guess then.

Comment (tcoratger) [reply tests/tests/debug_api.rs:79]: Yes I agree this is in the `v` field but here in fact the problem is just that the `transaction_by_hash` function will only search in the database without doing any calculation. However, in our database, we generate arbitrary transactions using the `arbitrary_with_optional_fields` function. And in this function, for Legacy transactions, we set the parity field.

I thought it was maybe useless and I tried to set it to `None` in this arbitrary transaction generation function like:

```rust
signature: Some(reth_rpc_types::Signature {
                r: transaction_signed.signature.r,
                s: transaction_signed.signature.s,
                v: if transaction_signed.is_legacy() {
                    U256::from(transaction_signed.signature.v(transaction_signed.chain_id()))
                } else {
                    U256::from(transaction_signed.signature.odd_y_parity)
                },
                y_parity: if transaction_signed.is_legacy() {None} else {Some(transaction_signed.signature.odd_y_parity.into()},
            }),
```



but in this case the `assert_eq!(transaction.from, transaction_signed.recover_signer().unwrap());` test no longer passes...

What we could do is at the end of the `transaction_by_hash` function, do an if Legacy, then set parity to None, but it's a bit overkilled right? I think we can say that field parity is "not annoying" right?

**Review by greged93 — APPROVED** [review]
